### ISSUE TITLE
Update tutorials so Floki does not parse documents multiple times

### ISSUE THREAD

**Issue by oltarasenko** [oltarasenko/crawly#42]
Based on:
https://elixirforum.com/t/web-scraping-tools/4823/31

```
Stop parsing each page four times.

When you run response.body |> Floki.find(...), you’re really running the equivalent of response.body |> Floki.parse() |> Floki.find(...) which means your four Floki.finds are parsing the whole document four times.

Instead, try parsed_body = Floki.parse(response.body) then parsed_body |> Floki.find(...).```

**Comment by philss** [comment]
For that I recommend that we start to use the newer functions [`Floki.parse_document/1`](https://hexdocs.pm/floki/Floki.html#parse_document/1) or [`Floki.parse_fragment/1`](https://hexdocs.pm/floki/Floki.html#parse_fragment/1).

They are functions that can return errors in case of something goes wrong with the parsing. Also, [`Floki.parse/1`](https://hexdocs.pm/floki/Floki.html#parse/1) is deprecated now.

**TLDR**:

```elixir
with {:ok, parsed_body} <- Floki.parse_document(response.body) do
  # Floki.find(parsed_body, "a.foo") or something similar
end
```

**Comment by oltarasenko** [comment]
I think this issue is done now
